### PR TITLE
BPM-109 Display active time event

### PR DESCRIPTION
**Pull request by epaltas**
#### What's this PR for?
Display active time-event on the left
#### Where should the reviewer start?
On the left side is displaying the currently active time event
#### Checklist
- [ ] Green travis build
- [ ] Newman tests (if required)
- [ ] Unit test (always)

### CODE REVIEW

**Review by oemel09 — CHANGES_REQUESTED** [review]
- Your branch seems to be many commits behind master, try to rebase it first or merge master into it.
- The build fails because some tests are not passing. Please fix the tests and make sure all new code has tests as well
- When I open the project in webstorm, eslint reports many errors. Once you have webstorm have a look at it and try to fix them.

File: src/component/time-active/TimeActive.jsx [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+      skills,
+      templateName,
+      organization_name: organizationName,
+      project_name: projectName,
+      activity,
+      template_id: templateId,
+      stop_time: stopTime,
+    } = this.props.activeTimeEvent;
+
+    console.log('activeTimeEvent',this.props.activeTimeEvent)
```
Comment (enriquezrene) [inline src/component/time-active/TimeActive.jsx]: we don't want to log anything. Please remove the console.log

**Review by enriquezrene — CHANGES_REQUESTED** [review]

File: src/component/time-active/TimeActive.jsx (hunk 1 above)
Comment (epaltas) [reply src/component/time-active/TimeActive.jsx]: Done

File: src/component/time-event/time-event-actions.js [hunk 2]
```diff
@@ -2,19 +2,45 @@ import axios from 'axios';
...
   timeEvent,
+  templateName,
+});
+
+export const setActiveTimeEventDuration = duration => ({
+  type: SET_ACTIVE_TIME_EVENT_DURATION,
+  duration,
+});
+
+export const getLastActiveTime = () => ((dispatch, getState) => {
```
Comment (oemel09) [inline src/component/time-event/time-event-actions.js:29]: This method is not used anywhere. But we should call it once the user gets logged in to display the active time event right away.
The call should happen in login-actions.js in the performLogin method after loading all the time-templates.

Also make sure to write tests for this method. Codecov is failing due to this

File: src/component/time-event/time-event-reducer.js [hunk 3]
```diff
@@ -1,9 +1,30 @@
...
 const timeEventReducer = (state = {}, action) => {
   switch (action.type) {
     case SET_ACTIVE_TIME_EVENT:
-      return action.timeEvent;
+      return {
+        ...action.timeEvent,
+        duration: initializeDuration(action.timeEvent.start_time, action.timeEvent.stop_time),
+        templateName: action.templateName,
+      };
+    case SET_ACTIVE_TIME_EVENT_DURATION:
```
Comment (oemel09) [inline src/component/time-event/time-event-reducer.js:23]: You added this case and it's also missing tests, please add a test for this case.

File: src/component/time-active/TimeActive.jsx [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+                <Typography className={classes.activity} variant="body1">
+                  {`${TemplateConst.ACTIVITY}: ${activity}`}
+                </Typography>
+                <Typography variant="h3">
+                  {this.formatDuration(this.props.activeTimeEvent.duration)}
+                </Typography>
+                <Typography className={classes.skills} variant="body1">
+                  {`${TemplateConst.SKILLS}: ${this.getSkillsList(skills)}`}
+                </Typography>
+                <TextField label="What did you do?" placeholder="Implemented some changes" />
```
Comment (oemel09) [inline src/component/time-active/TimeActive.jsx]: Please create a constants file for these two strings and use them here

**Review by oemel09 — CHANGES_REQUESTED** [review]

File: src/component/time-active/TimeActive.jsx (hunk 4 above)
Comment (epaltas) [reply src/component/time-active/TimeActive.jsx]: Done

File: src/component/time-event/time-event-actions.js (hunk 2 above)
Comment (epaltas) [reply src/component/time-event/time-event-actions.js:29]: done

File: src/component/time-event/time-event-reducer.js (hunk 3 above)
Comment (epaltas) [reply src/component/time-event/time-event-reducer.js:23]: Done

**Review by oemel09 — APPROVED** [review]
SonarCloud fails, but I think we should merge this code and create a ticket to fix all the errors found by SonarCloud in a separate PR.

**Review by enriquezrene — APPROVED** [review]

**Pull request merged**